### PR TITLE
add C implementation for `@stdlib/math/base/special/frexp`

### LINKED ISSUE #637: [RFC]: add C implementation for `@stdlib/math/base/special/frexp`
### Description

This RFC proposes adding a C implementation to [`@stdlib/math/base/special/frexp`](https://github.com/stdlib-js/stdlib/tree/b43a75923e47775446d3b6d7219ee42e07f7ce63/lib/node_modules/%40stdlib/math/base/special/frexp).

Having a C implementation will allow us to implement other special functions in C.

```c
void stdlib_base_frexp( const double x, double *frac, int32_t *exp );
```

### Related Issues

None.

### Questions

- The proposed API differs from [C99](https://en.cppreference.com/w/c/numeric/math/frexp). Is this desirable?

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Pranavchiku**
Resolves #637 

## Checklist

- [x] update readme.md
- [x] include.gypi
- [x] binding.gyp
- [x] include/stdlib/math/base/special/
- [ ] manifest.json
- [ ] src
- [ ] lib
- [ ] examples
- [ ] benchmark
- [ ] test
* * *

@kgryte

### CODE REVIEW

File: lib/node_modules/@stdlib/math/base/special/frexp/README.md [hunk 1]
```diff
@@ -168,6 +168,98 @@ for ( i = 0; i < 100; i++ ) {
...
+#include <stdint.h>
+
+stdlib_base_frexp( 4.0, &frac, &exp );
+```
+
+The function accepts the following arguments:
+
+-   **x**: `[in] float` input value.
+-   **frac**: `[out] float*` destination to a normalized fraction.
+-   **exp**: `[out] int32_t*` destination to an integer power of two.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/frexp/README.md]:
```suggestion
-   **frac**: `[out] float*` destination for the normalized fraction.
-   **exp**: `[out] int32_t*` destination for the integer power of two.
```

File: lib/node_modules/@stdlib/math/base/special/frexp/include/stdlib/math/base/special/frexp.h [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+* If C++, prevent name mangling so that the compiler emits a binary file having undecorated names, thus mirroring the behavior of a C compiler.
+*/
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+* Splits a double-precision floating-point number into a normalized fraction and an integer power of two.
+* The normalized fraction and exponent satisfy the relation x = frac * 2^exp.
+*/
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/frexp/include/stdlib/math/base/special/frexp.h]:
```suggestion
*/
```

File: lib/node_modules/@stdlib/math/base/special/frexp/include/stdlib/math/base/special/frexp.h [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#ifndef STDLIB_MATH_BASE_SPECIAL_FREXP_H
+#define STDLIB_MATH_BASE_SPECIAL_FREXP_H
+
+#include <stdint.h>
+
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/frexp/include/stdlib/math/base/special/frexp.h]:
```suggestion
```

**Comment by Pranavchiku** [comment]
Closed in reference to #651

**Pull request closed without merging**